src/search/utils/tuples.h

```diff
@@ -0,0 +1,31 @@
+#ifndef UTILS_TUPLES_H
```

Comment:
I wonder if adding some comments on how this template magic works would be beneficial. I only understood the code when Florian walked me through it. On the other hand I'm not sure if we need a detailed explanation, or are just happy that it does work ;).

Reply:
I think this is hard to understand and I also only understood it when Florian explained it to me.

A permanent place to find an explanation of it would be nice but the comments might not be the right place for it, is it?